In [41]:
from xgboost import XGBClassifier
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
from sklearn.metrics import roc_auc_score, f1_score, recall_score, precision_score, ConfusionMatrixDisplay, RocCurveDisplay
from sklearn.model_selection import StratifiedKFold, train_test_split
from imblearn.over_sampling import SMOTE, ADASYN, SMOTENC
import shap
import optuna


In [42]:
df = pd.read_csv("data/train-cat-encoded.csv")
print("Training set size: ", df.shape)
y = df['Will_Buy_EV']
x = df.drop(columns=['Will_Buy_EV'])
x = x.drop(columns=['id'])
binary_features = ['Home_Charging_Possible', 'Subsidy_Available', 
                     'City_Type_Urban', 'City_Type_Suburban', 'City_Type_Rural', 
                     'Current_Car_Type_Sedan', 'Current_Car_Type_SUV', 
                     'Current_Car_Type_Hatchback', 'Current_Car_Type_Truck', 
                     'Gender_Male', 'Gender_Female', 'Gender_Other']
print(binary_features)


Training set size:  (668665, 22)
['Home_Charging_Possible', 'Subsidy_Available', 'City_Type_Urban', 'City_Type_Suburban', 'City_Type_Rural', 'Current_Car_Type_Sedan', 'Current_Car_Type_SUV', 'Current_Car_Type_Hatchback', 'Current_Car_Type_Truck', 'Gender_Male', 'Gender_Female', 'Gender_Other']


In [43]:
n_splits = 5
skf = StratifiedKFold(n_splits=n_splits, shuffle=True, random_state=42)
subsetNum = 50000
subset_x = x.iloc[:subsetNum]
subset_y = y.iloc[:subsetNum]
smote = SMOTENC(
    random_state=42,
    categorical_features=binary_features)
ada = ADASYN(n_neighbors=5, random_state=42)


In [44]:
hyperparameters = {
    'n_estimators': [50, 100, 200, 300],
    'max_depth': [4, 6, 8, 10],
    'learning_rate': [0.02, 0.05, 0.1, 0.15],
    'subsample': [0.75, 0.8, 0.85],
    'colsample_bytree': [0.75, 0.8, 0.85]
}


In [45]:
X_train, X_test, y_train, y_test = train_test_split(
    subset_x, subset_y, test_size=0.2, random_state=42
)
X_train_resampled, y_train_resampled = smote.fit_resample(X_train, y_train)


In [46]:
print("==========XGBoost Cross Validation Training Loop==========")

def objective(trial):
    n_estimators = trial.suggest_int("n_estimators", 50, 300, step=50)
    max_depth = trial.suggest_categorical("max_depth", [4, 6, 8, 10])
    learning_rate = trial.suggest_categorical("learning_rate", [0.05, 0.1, 0.15])
    subsample = trial.suggest_categorical("subsample", [0.75, 0.8, 0.85])
    colsample_bytree = trial.suggest_categorical("colsample_bytree", [0.75, 0.8, 0.85])
    
    print(f"\n========== n_estimators {n_estimators} max_depth {max_depth}==========")
        
    # 1. Split data (using .iloc if X/y are pandas DataFrames)

    # X_train_resampled, y_train_resampled = ada.fit_resample(X_train, y_train)

    model = XGBClassifier(
    n_estimators=n_estimators,      # number of boosting rounds (trees)
    max_depth=max_depth,           # tree depth
    learning_rate=learning_rate,     # shrinkage per round
    subsample=subsample,         # row sampling per tree
    colsample_bytree=colsample_bytree,  # feature sampling per tree
    eval_metric='logloss'
)
    model.fit(X_train_resampled, y_train_resampled)
    y_pred = model.predict(X_test)
    y_prob = model.predict_proba(X_test)[:, 1]
    
    auc = roc_auc_score(y_test, y_prob)
    f1 = f1_score(y_test, y_pred)
    recall = recall_score(y_test, y_pred)
    precision = precision_score(y_test, y_pred)

    print(f"AUC: {auc:.4f}")
    print(f"F1: {f1:.4f} | Recall: {recall:.4f} | Precision: {precision:.4f}")

    return auc


==========XGBoost Cross Validation Training Loop==========


In [47]:
study = optuna.create_study(direction="maximize")
study.optimize(objective, n_trials=100)


[I 2026-09-09 23:16:25,530] A new study created in memory with name: no-name-5291587b-2c1d-4341-9d64-3a6565accefa



========== n_estimators 200 max_depth 4==========


[I 2026-09-09 23:16:26,657] Trial 0 finished with value: 0.9367888732798522 and parameters: {'n_estimators': 200, 'max_depth': 4, 'learning_rate': 0.05, 'subsample': 0.8, 'colsample_bytree': 0.75}. Best is trial 0 with value: 0.9367888732798522.


AUC: 0.9368
F1: 0.7009 | Recall: 0.7717 | Precision: 0.6420

========== n_estimators 150 max_depth 10==========


[I 2026-09-09 23:16:28,049] Trial 1 finished with value: 0.9342670794682609 and parameters: {'n_estimators': 150, 'max_depth': 10, 'learning_rate': 0.1, 'subsample': 0.8, 'colsample_bytree': 0.8}. Best is trial 0 with value: 0.9367888732798522.


AUC: 0.9343
F1: 0.6858 | Recall: 0.7131 | Precision: 0.6605

========== n_estimators 50 max_depth 4==========


[I 2026-09-09 23:16:28,409] Trial 2 finished with value: 0.9367441785259571 and parameters: {'n_estimators': 50, 'max_depth': 4, 'learning_rate': 0.1, 'subsample': 0.8, 'colsample_bytree': 0.75}. Best is trial 0 with value: 0.9367888732798522.


AUC: 0.9367
F1: 0.6963 | Recall: 0.7980 | Precision: 0.6176

========== n_estimators 50 max_depth 10==========


[I 2026-09-09 23:16:28,957] Trial 3 finished with value: 0.9357541508928782 and parameters: {'n_estimators': 50, 'max_depth': 10, 'learning_rate': 0.1, 'subsample': 0.8, 'colsample_bytree': 0.75}. Best is trial 0 with value: 0.9367888732798522.


AUC: 0.9358
F1: 0.6938 | Recall: 0.7395 | Precision: 0.6534

========== n_estimators 200 max_depth 6==========


[I 2026-09-09 23:16:30,039] Trial 4 finished with value: 0.936320143226563 and parameters: {'n_estimators': 200, 'max_depth': 6, 'learning_rate': 0.1, 'subsample': 0.8, 'colsample_bytree': 0.85}. Best is trial 0 with value: 0.9367888732798522.


AUC: 0.9363
F1: 0.6936 | Recall: 0.7289 | Precision: 0.6615

========== n_estimators 150 max_depth 6==========


[I 2026-09-09 23:16:31,055] Trial 5 finished with value: 0.9368427470511347 and parameters: {'n_estimators': 150, 'max_depth': 6, 'learning_rate': 0.05, 'subsample': 0.85, 'colsample_bytree': 0.85}. Best is trial 5 with value: 0.9368427470511347.


AUC: 0.9368
F1: 0.7003 | Recall: 0.7605 | Precision: 0.6489

========== n_estimators 100 max_depth 4==========


[I 2026-09-09 23:16:31,600] Trial 6 finished with value: 0.9373956769369985 and parameters: {'n_estimators': 100, 'max_depth': 4, 'learning_rate': 0.15, 'subsample': 0.8, 'colsample_bytree': 0.8}. Best is trial 6 with value: 0.9373956769369985.


AUC: 0.9374
F1: 0.6992 | Recall: 0.7512 | Precision: 0.6539

========== n_estimators 100 max_depth 4==========


[I 2026-09-09 23:16:32,185] Trial 7 finished with value: 0.9371409945084057 and parameters: {'n_estimators': 100, 'max_depth': 4, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.75}. Best is trial 6 with value: 0.9373956769369985.


AUC: 0.9371
F1: 0.7010 | Recall: 0.7570 | Precision: 0.6527

========== n_estimators 100 max_depth 4==========


[I 2026-09-09 23:16:32,717] Trial 8 finished with value: 0.9363064100043946 and parameters: {'n_estimators': 100, 'max_depth': 4, 'learning_rate': 0.1, 'subsample': 0.8, 'colsample_bytree': 0.8}. Best is trial 6 with value: 0.9373956769369985.


AUC: 0.9363
F1: 0.6985 | Recall: 0.7705 | Precision: 0.6388

========== n_estimators 250 max_depth 4==========


[I 2026-09-09 23:16:34,520] Trial 9 finished with value: 0.9369159320628444 and parameters: {'n_estimators': 250, 'max_depth': 4, 'learning_rate': 0.05, 'subsample': 0.85, 'colsample_bytree': 0.75}. Best is trial 6 with value: 0.9373956769369985.


AUC: 0.9369
F1: 0.7010 | Recall: 0.7617 | Precision: 0.6492

========== n_estimators 300 max_depth 4==========


[I 2026-09-09 23:16:36,196] Trial 10 finished with value: 0.9376066531212612 and parameters: {'n_estimators': 300, 'max_depth': 4, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 10 with value: 0.9376066531212612.


AUC: 0.9376
F1: 0.6956 | Recall: 0.7324 | Precision: 0.6623

========== n_estimators 200 max_depth 4==========


[I 2026-09-09 23:16:37,214] Trial 11 finished with value: 0.9370891330601658 and parameters: {'n_estimators': 200, 'max_depth': 4, 'learning_rate': 0.15, 'subsample': 0.85, 'colsample_bytree': 0.8}. Best is trial 10 with value: 0.9376066531212612.


AUC: 0.9371
F1: 0.6968 | Recall: 0.7395 | Precision: 0.6588

========== n_estimators 300 max_depth 4==========


[I 2026-09-09 23:16:38,891] Trial 12 finished with value: 0.9376066531212612 and parameters: {'n_estimators': 300, 'max_depth': 4, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 10 with value: 0.9376066531212612.


AUC: 0.9376
F1: 0.6956 | Recall: 0.7324 | Precision: 0.6623

========== n_estimators 300 max_depth 4==========


[I 2026-09-09 23:16:40,589] Trial 13 finished with value: 0.9376066531212612 and parameters: {'n_estimators': 300, 'max_depth': 4, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 10 with value: 0.9376066531212612.


AUC: 0.9376
F1: 0.6956 | Recall: 0.7324 | Precision: 0.6623

========== n_estimators 300 max_depth 6==========


[I 2026-09-09 23:16:42,434] Trial 14 finished with value: 0.9341271347570131 and parameters: {'n_estimators': 300, 'max_depth': 6, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.85}. Best is trial 10 with value: 0.9376066531212612.


AUC: 0.9341
F1: 0.6889 | Recall: 0.7137 | Precision: 0.6658

========== n_estimators 300 max_depth 8==========


[I 2026-09-09 23:16:44,700] Trial 15 finished with value: 0.9297720440457269 and parameters: {'n_estimators': 300, 'max_depth': 8, 'learning_rate': 0.15, 'subsample': 0.8, 'colsample_bytree': 0.8}. Best is trial 10 with value: 0.9376066531212612.


AUC: 0.9298
F1: 0.6665 | Recall: 0.6862 | Precision: 0.6479

========== n_estimators 300 max_depth 10==========


[I 2026-09-09 23:16:47,670] Trial 16 finished with value: 0.9298765789322062 and parameters: {'n_estimators': 300, 'max_depth': 10, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.75}. Best is trial 10 with value: 0.9376066531212612.


AUC: 0.9299
F1: 0.6756 | Recall: 0.6915 | Precision: 0.6605

========== n_estimators 300 max_depth 8==========


[I 2026-09-09 23:16:50,006] Trial 17 finished with value: 0.9333368919677667 and parameters: {'n_estimators': 300, 'max_depth': 8, 'learning_rate': 0.1, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 10 with value: 0.9376066531212612.


AUC: 0.9333
F1: 0.6863 | Recall: 0.7096 | Precision: 0.6645

========== n_estimators 200 max_depth 8==========


[I 2026-09-09 23:16:51,383] Trial 18 finished with value: 0.9319420343639815 and parameters: {'n_estimators': 200, 'max_depth': 8, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 10 with value: 0.9376066531212612.


AUC: 0.9319
F1: 0.6805 | Recall: 0.7084 | Precision: 0.6548

========== n_estimators 250 max_depth 4==========


[I 2026-09-09 23:16:52,517] Trial 19 finished with value: 0.9368925608724189 and parameters: {'n_estimators': 250, 'max_depth': 4, 'learning_rate': 0.05, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 10 with value: 0.9376066531212612.


AUC: 0.9369
F1: 0.7002 | Recall: 0.7611 | Precision: 0.6484

========== n_estimators 250 max_depth 6==========


[I 2026-09-09 23:16:54,497] Trial 20 finished with value: 0.934252534256093 and parameters: {'n_estimators': 250, 'max_depth': 6, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 10 with value: 0.9376066531212612.


AUC: 0.9343
F1: 0.6892 | Recall: 0.7213 | Precision: 0.6599

========== n_estimators 300 max_depth 4==========


[I 2026-09-09 23:16:56,109] Trial 21 finished with value: 0.9376066531212612 and parameters: {'n_estimators': 300, 'max_depth': 4, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 10 with value: 0.9376066531212612.


AUC: 0.9376
F1: 0.6956 | Recall: 0.7324 | Precision: 0.6623

========== n_estimators 250 max_depth 4==========


[I 2026-09-09 23:16:57,299] Trial 22 finished with value: 0.9372879293944334 and parameters: {'n_estimators': 250, 'max_depth': 4, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.75}. Best is trial 10 with value: 0.9376066531212612.


AUC: 0.9373
F1: 0.6950 | Recall: 0.7313 | Precision: 0.6622

========== n_estimators 200 max_depth 4==========


[I 2026-09-09 23:16:58,267] Trial 23 finished with value: 0.9375296905908577 and parameters: {'n_estimators': 200, 'max_depth': 4, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.85}. Best is trial 10 with value: 0.9376066531212612.


AUC: 0.9375
F1: 0.7016 | Recall: 0.7477 | Precision: 0.6610

========== n_estimators 250 max_depth 4==========


[I 2026-09-09 23:16:59,431] Trial 24 finished with value: 0.9378071440433544 and parameters: {'n_estimators': 250, 'max_depth': 4, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 24 with value: 0.9378071440433544.


AUC: 0.9378
F1: 0.6965 | Recall: 0.7330 | Precision: 0.6635

========== n_estimators 250 max_depth 4==========


[I 2026-09-09 23:17:00,608] Trial 25 finished with value: 0.9372141795201152 and parameters: {'n_estimators': 250, 'max_depth': 4, 'learning_rate': 0.15, 'subsample': 0.8, 'colsample_bytree': 0.8}. Best is trial 24 with value: 0.9378071440433544.


AUC: 0.9372
F1: 0.6956 | Recall: 0.7278 | Precision: 0.6661

========== n_estimators 300 max_depth 4==========


[I 2026-09-09 23:17:02,843] Trial 26 finished with value: 0.9369902114958578 and parameters: {'n_estimators': 300, 'max_depth': 4, 'learning_rate': 0.15, 'subsample': 0.85, 'colsample_bytree': 0.85}. Best is trial 24 with value: 0.9378071440433544.


AUC: 0.9370
F1: 0.6938 | Recall: 0.7295 | Precision: 0.6614

========== n_estimators 250 max_depth 4==========


[I 2026-09-09 23:17:03,981] Trial 27 finished with value: 0.9378071440433544 and parameters: {'n_estimators': 250, 'max_depth': 4, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 24 with value: 0.9378071440433544.


AUC: 0.9378
F1: 0.6965 | Recall: 0.7330 | Precision: 0.6635

========== n_estimators 250 max_depth 4==========


[I 2026-09-09 23:17:04,968] Trial 28 finished with value: 0.9378071440433544 and parameters: {'n_estimators': 250, 'max_depth': 4, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 24 with value: 0.9378071440433544.


AUC: 0.9378
F1: 0.6965 | Recall: 0.7330 | Precision: 0.6635

========== n_estimators 200 max_depth 6==========


[I 2026-09-09 23:17:06,109] Trial 29 finished with value: 0.9363570711195915 and parameters: {'n_estimators': 200, 'max_depth': 6, 'learning_rate': 0.1, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 24 with value: 0.9378071440433544.


AUC: 0.9364
F1: 0.6957 | Recall: 0.7342 | Precision: 0.6610

========== n_estimators 150 max_depth 4==========


[I 2026-09-09 23:17:06,886] Trial 30 finished with value: 0.9378306564494319 and parameters: {'n_estimators': 150, 'max_depth': 4, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 30 with value: 0.9378306564494319.


AUC: 0.9378
F1: 0.7005 | Recall: 0.7430 | Precision: 0.6627

========== n_estimators 150 max_depth 4==========


[I 2026-09-09 23:17:07,701] Trial 31 finished with value: 0.9378306564494319 and parameters: {'n_estimators': 150, 'max_depth': 4, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 30 with value: 0.9378306564494319.


AUC: 0.9378
F1: 0.7005 | Recall: 0.7430 | Precision: 0.6627

========== n_estimators 150 max_depth 4==========


[I 2026-09-09 23:17:08,385] Trial 32 finished with value: 0.9378306564494319 and parameters: {'n_estimators': 150, 'max_depth': 4, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 30 with value: 0.9378306564494319.


AUC: 0.9378
F1: 0.7005 | Recall: 0.7430 | Precision: 0.6627

========== n_estimators 150 max_depth 4==========


[I 2026-09-09 23:17:09,170] Trial 33 finished with value: 0.9378306564494319 and parameters: {'n_estimators': 150, 'max_depth': 4, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 30 with value: 0.9378306564494319.


AUC: 0.9378
F1: 0.7005 | Recall: 0.7430 | Precision: 0.6627

========== n_estimators 100 max_depth 8==========


[I 2026-09-09 23:17:10,762] Trial 34 finished with value: 0.9351166328313965 and parameters: {'n_estimators': 100, 'max_depth': 8, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 30 with value: 0.9378306564494319.


AUC: 0.9351
F1: 0.6865 | Recall: 0.7242 | Precision: 0.6524

========== n_estimators 150 max_depth 4==========


[I 2026-09-09 23:17:12,129] Trial 35 finished with value: 0.9368467716972201 and parameters: {'n_estimators': 150, 'max_depth': 4, 'learning_rate': 0.05, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 30 with value: 0.9378306564494319.


AUC: 0.9368
F1: 0.6998 | Recall: 0.7875 | Precision: 0.6297

========== n_estimators 150 max_depth 4==========


[I 2026-09-09 23:17:13,459] Trial 36 finished with value: 0.9377374541190346 and parameters: {'n_estimators': 150, 'max_depth': 4, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.85}. Best is trial 30 with value: 0.9378306564494319.


AUC: 0.9377
F1: 0.7014 | Recall: 0.7488 | Precision: 0.6596

========== n_estimators 150 max_depth 4==========


[I 2026-09-09 23:17:14,359] Trial 37 finished with value: 0.9378306564494319 and parameters: {'n_estimators': 150, 'max_depth': 4, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 30 with value: 0.9378306564494319.


AUC: 0.9378
F1: 0.7005 | Recall: 0.7430 | Precision: 0.6627

========== n_estimators 150 max_depth 10==========


[I 2026-09-09 23:17:15,777] Trial 38 finished with value: 0.931197474838195 and parameters: {'n_estimators': 150, 'max_depth': 10, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 30 with value: 0.9378306564494319.


AUC: 0.9312
F1: 0.6803 | Recall: 0.7096 | Precision: 0.6534

========== n_estimators 50 max_depth 4==========


[I 2026-09-09 23:17:16,060] Trial 39 finished with value: 0.9367949808568062 and parameters: {'n_estimators': 50, 'max_depth': 4, 'learning_rate': 0.1, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 30 with value: 0.9378306564494319.


AUC: 0.9368
F1: 0.6967 | Recall: 0.7957 | Precision: 0.6197

========== n_estimators 100 max_depth 4==========


[I 2026-09-09 23:17:16,505] Trial 40 finished with value: 0.9370957701958152 and parameters: {'n_estimators': 100, 'max_depth': 4, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 30 with value: 0.9378306564494319.


AUC: 0.9371
F1: 0.6986 | Recall: 0.7512 | Precision: 0.6529

========== n_estimators 150 max_depth 4==========


[I 2026-09-09 23:17:17,183] Trial 41 finished with value: 0.9378306564494319 and parameters: {'n_estimators': 150, 'max_depth': 4, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 30 with value: 0.9378306564494319.


AUC: 0.9378
F1: 0.7005 | Recall: 0.7430 | Precision: 0.6627

========== n_estimators 50 max_depth 4==========


[I 2026-09-09 23:17:17,466] Trial 42 finished with value: 0.9367093688677103 and parameters: {'n_estimators': 50, 'max_depth': 4, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 30 with value: 0.9378306564494319.


AUC: 0.9367
F1: 0.7008 | Recall: 0.7804 | Precision: 0.6360

========== n_estimators 150 max_depth 4==========


[I 2026-09-09 23:17:18,097] Trial 43 finished with value: 0.9376083830129996 and parameters: {'n_estimators': 150, 'max_depth': 4, 'learning_rate': 0.15, 'subsample': 0.8, 'colsample_bytree': 0.8}. Best is trial 30 with value: 0.9378306564494319.


AUC: 0.9376
F1: 0.7012 | Recall: 0.7441 | Precision: 0.6630

========== n_estimators 200 max_depth 4==========


[I 2026-09-09 23:17:18,958] Trial 44 finished with value: 0.9378566401294213 and parameters: {'n_estimators': 200, 'max_depth': 4, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9379
F1: 0.7005 | Recall: 0.7430 | Precision: 0.6627

========== n_estimators 100 max_depth 10==========


[I 2026-09-09 23:17:19,875] Trial 45 finished with value: 0.933198500628692 and parameters: {'n_estimators': 100, 'max_depth': 10, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9332
F1: 0.6810 | Recall: 0.7137 | Precision: 0.6512

========== n_estimators 100 max_depth 4==========


[I 2026-09-09 23:17:21,418] Trial 46 finished with value: 0.9367034731142344 and parameters: {'n_estimators': 100, 'max_depth': 4, 'learning_rate': 0.05, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9367
F1: 0.6962 | Recall: 0.7957 | Precision: 0.6189

========== n_estimators 200 max_depth 4==========


[I 2026-09-09 23:17:23,205] Trial 47 finished with value: 0.9371084089966798 and parameters: {'n_estimators': 200, 'max_depth': 4, 'learning_rate': 0.1, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9371
F1: 0.7010 | Recall: 0.7488 | Precision: 0.6589

========== n_estimators 200 max_depth 4==========


[I 2026-09-09 23:17:24,448] Trial 48 finished with value: 0.9375943320556142 and parameters: {'n_estimators': 200, 'max_depth': 4, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.75}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9376
F1: 0.7025 | Recall: 0.7430 | Precision: 0.6661

========== n_estimators 200 max_depth 10==========


[I 2026-09-09 23:17:26,344] Trial 49 finished with value: 0.9297415414648695 and parameters: {'n_estimators': 200, 'max_depth': 10, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9297
F1: 0.6772 | Recall: 0.7014 | Precision: 0.6546

========== n_estimators 100 max_depth 6==========


[I 2026-09-09 23:17:27,136] Trial 50 finished with value: 0.9364752686204135 and parameters: {'n_estimators': 100, 'max_depth': 6, 'learning_rate': 0.15, 'subsample': 0.85, 'colsample_bytree': 0.85}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9365
F1: 0.6956 | Recall: 0.7371 | Precision: 0.6585

========== n_estimators 150 max_depth 4==========


[I 2026-09-09 23:17:27,897] Trial 51 finished with value: 0.9374823480434854 and parameters: {'n_estimators': 150, 'max_depth': 4, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.75}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9375
F1: 0.7017 | Recall: 0.7459 | Precision: 0.6625

========== n_estimators 150 max_depth 4==========


[I 2026-09-09 23:17:28,787] Trial 52 finished with value: 0.9369693821871707 and parameters: {'n_estimators': 150, 'max_depth': 4, 'learning_rate': 0.15, 'subsample': 0.85, 'colsample_bytree': 0.8}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9370
F1: 0.6965 | Recall: 0.7418 | Precision: 0.6565

========== n_estimators 150 max_depth 6==========


[I 2026-09-09 23:17:29,832] Trial 53 finished with value: 0.9368279194076625 and parameters: {'n_estimators': 150, 'max_depth': 6, 'learning_rate': 0.1, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9368
F1: 0.7022 | Recall: 0.7447 | Precision: 0.6642

========== n_estimators 200 max_depth 4==========


[I 2026-09-09 23:17:30,912] Trial 54 finished with value: 0.9378566401294213 and parameters: {'n_estimators': 200, 'max_depth': 4, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9379
F1: 0.7005 | Recall: 0.7430 | Precision: 0.6627

========== n_estimators 200 max_depth 4==========


[I 2026-09-09 23:17:31,970] Trial 55 finished with value: 0.9378566401294213 and parameters: {'n_estimators': 200, 'max_depth': 4, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9379
F1: 0.7005 | Recall: 0.7430 | Precision: 0.6627

========== n_estimators 200 max_depth 4==========


[I 2026-09-09 23:17:33,132] Trial 56 finished with value: 0.9373312472957203 and parameters: {'n_estimators': 200, 'max_depth': 4, 'learning_rate': 0.15, 'subsample': 0.8, 'colsample_bytree': 0.8}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9373
F1: 0.6948 | Recall: 0.7324 | Precision: 0.6609

========== n_estimators 200 max_depth 4==========


[I 2026-09-09 23:17:34,119] Trial 57 finished with value: 0.9378566401294213 and parameters: {'n_estimators': 200, 'max_depth': 4, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9379
F1: 0.7005 | Recall: 0.7430 | Precision: 0.6627

========== n_estimators 200 max_depth 4==========


[I 2026-09-09 23:17:35,034] Trial 58 finished with value: 0.9378566401294213 and parameters: {'n_estimators': 200, 'max_depth': 4, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9379
F1: 0.7005 | Recall: 0.7430 | Precision: 0.6627

========== n_estimators 200 max_depth 4==========


[I 2026-09-09 23:17:35,902] Trial 59 finished with value: 0.9378566401294213 and parameters: {'n_estimators': 200, 'max_depth': 4, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9379
F1: 0.7005 | Recall: 0.7430 | Precision: 0.6627

========== n_estimators 200 max_depth 4==========


[I 2026-09-09 23:17:36,700] Trial 60 finished with value: 0.9378566401294213 and parameters: {'n_estimators': 200, 'max_depth': 4, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9379
F1: 0.7005 | Recall: 0.7430 | Precision: 0.6627

========== n_estimators 200 max_depth 4==========


[I 2026-09-09 23:17:37,590] Trial 61 finished with value: 0.9378566401294213 and parameters: {'n_estimators': 200, 'max_depth': 4, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9379
F1: 0.7005 | Recall: 0.7430 | Precision: 0.6627

========== n_estimators 200 max_depth 4==========


[I 2026-09-09 23:17:38,548] Trial 62 finished with value: 0.9378566401294213 and parameters: {'n_estimators': 200, 'max_depth': 4, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9379
F1: 0.7005 | Recall: 0.7430 | Precision: 0.6627

========== n_estimators 200 max_depth 4==========


[I 2026-09-09 23:17:39,402] Trial 63 finished with value: 0.9367183007577067 and parameters: {'n_estimators': 200, 'max_depth': 4, 'learning_rate': 0.05, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9367
F1: 0.6971 | Recall: 0.7681 | Precision: 0.6381

========== n_estimators 200 max_depth 6==========


[I 2026-09-09 23:17:40,412] Trial 64 finished with value: 0.9356810717929078 and parameters: {'n_estimators': 200, 'max_depth': 6, 'learning_rate': 0.15, 'subsample': 0.85, 'colsample_bytree': 0.8}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9357
F1: 0.6898 | Recall: 0.7213 | Precision: 0.6609

========== n_estimators 200 max_depth 4==========


[I 2026-09-09 23:17:41,204] Trial 65 finished with value: 0.9373547597017977 and parameters: {'n_estimators': 200, 'max_depth': 4, 'learning_rate': 0.1, 'subsample': 0.85, 'colsample_bytree': 0.8}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9374
F1: 0.6982 | Recall: 0.7436 | Precision: 0.6580

========== n_estimators 200 max_depth 6==========


[I 2026-09-09 23:17:42,386] Trial 66 finished with value: 0.9351126787931371 and parameters: {'n_estimators': 200, 'max_depth': 6, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9351
F1: 0.6901 | Recall: 0.7248 | Precision: 0.6585

========== n_estimators 200 max_depth 8==========


[I 2026-09-09 23:17:43,791] Trial 67 finished with value: 0.9362983254083109 and parameters: {'n_estimators': 200, 'max_depth': 8, 'learning_rate': 0.05, 'subsample': 0.75, 'colsample_bytree': 0.85}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9363
F1: 0.6949 | Recall: 0.7342 | Precision: 0.6597

========== n_estimators 250 max_depth 8==========


[I 2026-09-09 23:17:45,932] Trial 68 finished with value: 0.9308987331261416 and parameters: {'n_estimators': 250, 'max_depth': 8, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9309
F1: 0.6756 | Recall: 0.6967 | Precision: 0.6556

========== n_estimators 150 max_depth 8==========


[I 2026-09-09 23:17:47,538] Trial 69 finished with value: 0.9356324230007536 and parameters: {'n_estimators': 150, 'max_depth': 8, 'learning_rate': 0.1, 'subsample': 0.8, 'colsample_bytree': 0.85}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9356
F1: 0.6946 | Recall: 0.7289 | Precision: 0.6633

========== n_estimators 250 max_depth 4==========


[I 2026-09-09 23:17:50,375] Trial 70 finished with value: 0.9373301175705033 and parameters: {'n_estimators': 250, 'max_depth': 4, 'learning_rate': 0.1, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9373
F1: 0.6980 | Recall: 0.7400 | Precision: 0.6604

========== n_estimators 200 max_depth 4==========


[I 2026-09-09 23:17:52,702] Trial 71 finished with value: 0.9378566401294213 and parameters: {'n_estimators': 200, 'max_depth': 4, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9379
F1: 0.7005 | Recall: 0.7430 | Precision: 0.6627

========== n_estimators 200 max_depth 4==========


[I 2026-09-09 23:17:54,055] Trial 72 finished with value: 0.9378566401294213 and parameters: {'n_estimators': 200, 'max_depth': 4, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9379
F1: 0.7005 | Recall: 0.7430 | Precision: 0.6627

========== n_estimators 200 max_depth 4==========


[I 2026-09-09 23:17:55,376] Trial 73 finished with value: 0.9369524363089166 and parameters: {'n_estimators': 200, 'max_depth': 4, 'learning_rate': 0.15, 'subsample': 0.8, 'colsample_bytree': 0.85}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9370
F1: 0.7013 | Recall: 0.7436 | Precision: 0.6635

========== n_estimators 200 max_depth 8==========


[I 2026-09-09 23:17:57,292] Trial 74 finished with value: 0.9326443704097852 and parameters: {'n_estimators': 200, 'max_depth': 8, 'learning_rate': 0.15, 'subsample': 0.8, 'colsample_bytree': 0.75}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9326
F1: 0.6761 | Recall: 0.6979 | Precision: 0.6557

========== n_estimators 250 max_depth 10==========


[I 2026-09-09 23:18:00,591] Trial 75 finished with value: 0.9285851970975099 and parameters: {'n_estimators': 250, 'max_depth': 10, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9286
F1: 0.6740 | Recall: 0.6979 | Precision: 0.6517

========== n_estimators 150 max_depth 8==========


[I 2026-09-09 23:18:02,447] Trial 76 finished with value: 0.9332311214443312 and parameters: {'n_estimators': 150, 'max_depth': 8, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9332
F1: 0.6821 | Recall: 0.7143 | Precision: 0.6528

========== n_estimators 250 max_depth 6==========


[I 2026-09-09 23:18:03,928] Trial 77 finished with value: 0.9364044489708768 and parameters: {'n_estimators': 250, 'max_depth': 6, 'learning_rate': 0.1, 'subsample': 0.85, 'colsample_bytree': 0.85}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9364
F1: 0.6963 | Recall: 0.7242 | Precision: 0.6705

========== n_estimators 250 max_depth 6==========


[I 2026-09-09 23:18:05,359] Trial 78 finished with value: 0.935816568211114 and parameters: {'n_estimators': 250, 'max_depth': 6, 'learning_rate': 0.1, 'subsample': 0.75, 'colsample_bytree': 0.75}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9358
F1: 0.6918 | Recall: 0.7266 | Precision: 0.6601

========== n_estimators 200 max_depth 6==========


[I 2026-09-09 23:18:06,508] Trial 79 finished with value: 0.936707780191624 and parameters: {'n_estimators': 200, 'max_depth': 6, 'learning_rate': 0.05, 'subsample': 0.8, 'colsample_bytree': 0.8}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9367
F1: 0.6996 | Recall: 0.7512 | Precision: 0.6546

========== n_estimators 100 max_depth 10==========


[I 2026-09-09 23:18:07,719] Trial 80 finished with value: 0.9362780609622322 and parameters: {'n_estimators': 100, 'max_depth': 10, 'learning_rate': 0.05, 'subsample': 0.85, 'colsample_bytree': 0.75}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9363
F1: 0.6936 | Recall: 0.7389 | Precision: 0.6535

========== n_estimators 200 max_depth 8==========


[I 2026-09-09 23:18:09,237] Trial 81 finished with value: 0.9351514424896432 and parameters: {'n_estimators': 200, 'max_depth': 8, 'learning_rate': 0.1, 'subsample': 0.75, 'colsample_bytree': 0.75}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9352
F1: 0.6904 | Recall: 0.7196 | Precision: 0.6636

========== n_estimators 250 max_depth 4==========


[I 2026-09-09 23:18:10,413] Trial 82 finished with value: 0.9368028183254987 and parameters: {'n_estimators': 250, 'max_depth': 4, 'learning_rate': 0.15, 'subsample': 0.85, 'colsample_bytree': 0.8}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9368
F1: 0.6967 | Recall: 0.7371 | Precision: 0.6605

========== n_estimators 200 max_depth 4==========


[I 2026-09-09 23:18:11,384] Trial 83 finished with value: 0.9378566401294213 and parameters: {'n_estimators': 200, 'max_depth': 4, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9379
F1: 0.7005 | Recall: 0.7430 | Precision: 0.6627

========== n_estimators 200 max_depth 10==========


[I 2026-09-09 23:18:13,353] Trial 84 finished with value: 0.930517521473252 and parameters: {'n_estimators': 200, 'max_depth': 10, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.85}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9305
F1: 0.6763 | Recall: 0.7014 | Precision: 0.6529

========== n_estimators 200 max_depth 4==========


[I 2026-09-09 23:18:14,286] Trial 85 finished with value: 0.9375296905908577 and parameters: {'n_estimators': 200, 'max_depth': 4, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.85}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9375
F1: 0.7016 | Recall: 0.7477 | Precision: 0.6610

========== n_estimators 250 max_depth 4==========


[I 2026-09-09 23:18:15,443] Trial 86 finished with value: 0.9369606974245654 and parameters: {'n_estimators': 250, 'max_depth': 4, 'learning_rate': 0.05, 'subsample': 0.8, 'colsample_bytree': 0.85}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9370
F1: 0.7019 | Recall: 0.7629 | Precision: 0.6499

========== n_estimators 200 max_depth 4==========


[I 2026-09-09 23:18:16,410] Trial 87 finished with value: 0.9378566401294213 and parameters: {'n_estimators': 200, 'max_depth': 4, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9379
F1: 0.7005 | Recall: 0.7430 | Precision: 0.6627

========== n_estimators 50 max_depth 10==========


[I 2026-09-09 23:18:16,993] Trial 88 finished with value: 0.9358868936058682 and parameters: {'n_estimators': 50, 'max_depth': 10, 'learning_rate': 0.1, 'subsample': 0.85, 'colsample_bytree': 0.85}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9359
F1: 0.6970 | Recall: 0.7400 | Precision: 0.6587

========== n_estimators 150 max_depth 4==========


[I 2026-09-09 23:18:17,742] Trial 89 finished with value: 0.9369762664502114 and parameters: {'n_estimators': 150, 'max_depth': 4, 'learning_rate': 0.1, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9370
F1: 0.6976 | Recall: 0.7488 | Precision: 0.6529

========== n_estimators 150 max_depth 6==========


[I 2026-09-09 23:18:18,661] Trial 90 finished with value: 0.9363057392300471 and parameters: {'n_estimators': 150, 'max_depth': 6, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9363
F1: 0.6953 | Recall: 0.7342 | Precision: 0.6603

========== n_estimators 200 max_depth 4==========


[I 2026-09-09 23:18:19,669] Trial 91 finished with value: 0.9378566401294213 and parameters: {'n_estimators': 200, 'max_depth': 4, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9379
F1: 0.7005 | Recall: 0.7430 | Precision: 0.6627

========== n_estimators 200 max_depth 8==========


[I 2026-09-09 23:18:21,213] Trial 92 finished with value: 0.9336170638215666 and parameters: {'n_estimators': 200, 'max_depth': 8, 'learning_rate': 0.15, 'subsample': 0.85, 'colsample_bytree': 0.8}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9336
F1: 0.6888 | Recall: 0.7166 | Precision: 0.6631

========== n_estimators 200 max_depth 4==========


[I 2026-09-09 23:18:22,138] Trial 93 finished with value: 0.9370891330601658 and parameters: {'n_estimators': 200, 'max_depth': 4, 'learning_rate': 0.15, 'subsample': 0.85, 'colsample_bytree': 0.8}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9371
F1: 0.6968 | Recall: 0.7395 | Precision: 0.6588

========== n_estimators 200 max_depth 4==========


[I 2026-09-09 23:18:23,088] Trial 94 finished with value: 0.9378566401294213 and parameters: {'n_estimators': 200, 'max_depth': 4, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9379
F1: 0.7005 | Recall: 0.7430 | Precision: 0.6627

========== n_estimators 200 max_depth 4==========


[I 2026-09-09 23:18:24,080] Trial 95 finished with value: 0.9378566401294213 and parameters: {'n_estimators': 200, 'max_depth': 4, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9379
F1: 0.7005 | Recall: 0.7430 | Precision: 0.6627

========== n_estimators 200 max_depth 10==========


[I 2026-09-09 23:18:26,125] Trial 96 finished with value: 0.9334405442564205 and parameters: {'n_estimators': 200, 'max_depth': 10, 'learning_rate': 0.1, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9334
F1: 0.6877 | Recall: 0.7143 | Precision: 0.6630

========== n_estimators 200 max_depth 8==========


[I 2026-09-09 23:18:27,686] Trial 97 finished with value: 0.9365727780282004 and parameters: {'n_estimators': 200, 'max_depth': 8, 'learning_rate': 0.05, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9366
F1: 0.6970 | Recall: 0.7365 | Precision: 0.6614

========== n_estimators 200 max_depth 8==========


[I 2026-09-09 23:18:29,138] Trial 98 finished with value: 0.9319420343639815 and parameters: {'n_estimators': 200, 'max_depth': 8, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9319
F1: 0.6805 | Recall: 0.7084 | Precision: 0.6548

========== n_estimators 200 max_depth 4==========


[I 2026-09-09 23:18:30,173] Trial 99 finished with value: 0.9372432699444515 and parameters: {'n_estimators': 200, 'max_depth': 4, 'learning_rate': 0.1, 'subsample': 0.75, 'colsample_bytree': 0.85}. Best is trial 44 with value: 0.9378566401294213.


AUC: 0.9372
F1: 0.6981 | Recall: 0.7447 | Precision: 0.6570


In [48]:
print(f"Best Score: {study.best_value}")
print(f"Best Parameters: {study.best_params}")
best = study.best_trial

print(f"Trial Number: {best.number}")
print(f"Best Score: {best.value}")
print(f"Best Params: {best.params}")


Best Score: 0.9378566401294213
Best Parameters: {'n_estimators': 200, 'max_depth': 4, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.8}
Trial Number: 44
Best Score: 0.9378566401294213
Best Params: {'n_estimators': 200, 'max_depth': 4, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.8}


### Without oversampling (10000 samples)
- Avg AUC: 0.7961
- Avg F1: 0.6768
- Avg Recall: 0.6476
- Avg Precision: 0.7091
- Std AUC: 0.011227993955145489

### With SMOTE 
- (50000 samples)
- Avg AUC: 0.9380
- Avg F1: 0.6928
- Avg Recall: 0.6803
- Avg Precision: 0.7061
- Std AUC: 0.0027307035663074416

- (100000 samples)
- Avg AUC: 0.9366
- Avg F1: 0.7012
- Avg Recall: 0.7416
- Avg Precision: 0.6650
- Std AUC: 0.001856941113581715

(200000 samples)
- Avg AUC: 0.9399
- Avg F1: 0.6997
- Avg Recall: 0.6903
- Avg Precision: 0.7095
- Std AUC: 0.0008880157870868589

(500000 samples)
- Avg AUC: 0.9402
- Avg F1: 0.6996
- Avg Recall: 0.6896
- Avg Precision: 0.7098
- Std AUC: 0.0008942851500033605

### With ADASYN 
- (50000 samples)
- Avg AUC: 0.9379
- Avg F1: 0.6930
- Avg Recall: 0.6819
- Avg Precision: 0.7048
- Std AUC: 0.0026846216264028214